apps/web/components/thread/thread-item.tsx

```
@@ -0,0 +1,422 @@
+            {threadItem.query}
+          </div>
+        </div>
+      )}
+
+      <div className="flex w-full flex-col gap-4">
+
+        <GoalsRenderer goals={goalsWithSteps || []} reasoning={threadItem?.reasoning} />
+
+       {
+        toolCallAndResults?.map((toolCallAndResult) => (
+          <div className='flex flex-col gap-1'>
```

Biome (1.9.4)

[error] Missing key property for this element in iterable.

The order of the items may change, and having a key can help React identify which item was moved.
Check the React documentation. 

(lint/correctness/useJsxKeyInIterable)